pkg/collector/metric_collector.go

```diff
@@ -139,22 +138,12 @@ func (c *Collector) UpdateProcessEnergyUtilizationMetrics() {
 }
 
 func (c *Collector) updateResourceUtilizationMetrics() {
-	wg := &sync.WaitGroup{}
-	wg.Add(2)
-	go c.updateNodeResourceUtilizationMetrics(wg)
-	go c.updateProcessResourceUtilizationMetrics(wg)
-	wg.Wait()
+	c.updateProcessResourceUtilizationMetrics()
 	// aggregate processes' resource utilization metrics to containers, virtual machines and nodes
 	c.AggregateProcessResourceUtilizationMetrics()
 }
 
-// update the node metrics that are not related to aggregated resource utilization of processes
-func (c *Collector) updateNodeResourceUtilizationMetrics(wg *sync.WaitGroup) {
```

Comment:
Do we want to keep this for TODO list when we have metrics those are not related to aggregated resource utilization of processes. @rootfs